src/axom/sidre/core/Group.cpp

```diff
@@ -106,14 +112,15 @@ View* Group::getView( const std::string& path )
 
   if ( group == nullptr )
   {
-    SLIC_CHECK_MSG( group != nullptr,
-                    "Non-existent group in path " << path );
+    SLIC_CHECK_MSG(group != nullptr,
```

Comment:
Why is this check even here... we just checked if group == nullptr

Reply:
It looks to me like a style choice.  "If something is wrong at all, then do: check and report each specific failure mode."  Same pattern that @nselliott was commenting on at View.hpp around line 875.  I suppose we could take out the superfluous outer `if (  )` guards, if @kennyweiss really wants to.

Reply:
> Why is this check even here ... ?

I think it is there because it [can] show up in the SLIC error message (depending on the way we format the slic message).

Reply:
>  I suppose we could take out the superfluous outer if ( ) guards

The outer ``if`` is actually necessary, due to the ``return nullptr`` in line 118 (after the ``SLIC_CHECK_MSG``)

Reply:
My point is SLIC_CHECK_MSG runs the expression, which we have already established is false by the previous if.  It just adds confusion and we should put a SLIC_ERROR or SLIC_WARNING.  It just seems like an abuse of a check to me.

Reply:
Thanks for clarifying @white238.

The closest thing to ``SLIC_CHECK_MSG`` is either ``SLIC_WARNING``, but it has slightly different semantics, since we have the ``checksAreErrors`` flag:

https://github.com/LLNL/axom/blob/7c84b028e066622e5d61fbd631233faa7a042294/src/axom/slic/interface/slic.hpp#L27-L30

And we have different toggles to control whether codes should abort on errors and/or warnings: 

https://github.com/LLNL/axom/blob/7c84b028e066622e5d61fbd631233faa7a042294/src/axom/slic/interface/slic.hpp#L78-L82
and 
https://github.com/LLNL/axom/blob/7c84b028e066622e5d61fbd631233faa7a042294/src/axom/slic/interface/slic.hpp#L107-L111

Since we'd want consistent behavior throughout axom, perhaps we should leave it as is for this PR, create a separate issue and discuss it at our next team meeting?

Reply:
In this specific case, it is not entirely clear to me what the appropriate conversion might be, since, in addition to what I said above, the check is currently only run in debug mode, and ``SLIC_WARNING`` would add it in all configurations, but ``SLIC_DEBUG`` would add it to a different stream (e.g. ``std::out`` vs. ``std::err``).